### ISSUE TITLE
DataLoader with batch_size=1 fails silently

### ISSUE THREAD

**Issue by amrav** [fastai/fastai#1230]
https://github.com/fastai/fastai/blob/ac45d71d4fab9430f0c195aeb7f2ad7c823d43a1/fastai/basic_data.py#L80

Why is `skip_size1` set to `True`? When batch size is set to 1 on a `DataLoader` passed into a `DataBunch`, where labels are scalars, the effect of this is that the generator is exhausted because it skips over every element, so the training is never run, and then it fails with (what appear to be) unrelated errors.

If batch size should not be set to 1 for some reason, would be great to document and fail with a helpful error message.

**Comment by sgugger** [comment]
Pytorch will return an error with a batch size of 1 if you're using batchnorm in your model, so the option is there to skip the last batch if it's of size 1.
You're welcome to make a PR to suggest an appropriate warning or error message.
